catalog/ui/catalog-ui-search/src/main/webapp/component/session-timeout-modal/session-timeout.view.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/* jshint browser: true */
+/* global define */
+define([
+        'jquery',
+        'backbone',
+        'marionette',
+        'js/wreqr',
+        'js/CustomElements',
+        './session-timeout.hbs'
+    ],
+    function ($, Backbone, Marionette, wreqr, CustomElements, sessionTimeoutModalTemplate) {
+        var _this = null;
+        var timer = null;
+
+        var SessionTimeoutModal = Marionette.LayoutView.extend({
+            template: sessionTimeoutModalTemplate,
+            tagName: CustomElements.register('session-timeout-modal'),
+            model: null,
+
+            ui: {
+                continueWorkingBtn: "#continueWorkingBtn"
+            },
+
+            events: {
+                'click @ui.continueWorkingBtn': 'refreshSession'
+            },
+
+            initialize: function (options) {
+                _this = this;
```

Comment:
❓ Are you sure you need `_this`?  Marionette should be properly binding this.  Most uses of `_this` in this file could be replaced with `this` which should properly bind those methods as well to the proper context.  The `_this` in the `$.get` would need to use the context parameter.

http://api.jquery.com/jquery.ajax/

Reply:
In the places where `_this` is used, the scope is different than the `initialize` function, so I thought it was necessary. But I was not aware of the jQuery context parameter, or that Marionette affects the binding of `this`.